### ISSUE TITLE
faba/embed: differentiate components within each (gene, modality)

### ISSUE THREAD

**Issue by YPARK** [causalpathlab/asap-rs#19]
## Background

After the scalar-gate refactor (8a177bc), `faba rna-mod-embed` (alias `embed`) embeds every row of `gene_X/{m}/{detail}` for the same `(gene_X, m)` to the *same* tensor:

```
e_{g,m,d}  =  ρ_g · (1 + Σ_k z_g[k] · Q[k,m])     # detail `d` not in formula
b_{g,m,d}  =  b_comp[g, m]                          # detail `d` not in formula
```

i.e. components within `(g, m)` are completely interchangeable as far as the model is concerned. Their reads do contribute as separate positive draws to the NCE softmax (so a component with more counts pulls more gradient), but the model has no way to differentiate them in either the embedding direction or the bias magnitude.

This was a deliberate Path-1 simplification because **components are not identifiable across genes** — `m6A/comp_0` of gene X is the index-0 mixture component for gene X, and `m6A/comp_0` of gene Y is the index-0 mixture component for gene Y, but the two are biologically unrelated (per-gene mixture indices). So shared cross-gene parameters keyed on `(m, d)` (like the `ψ_{m, d}` we sketched but didn't wire) would learn meaningless averages.

## What's missing

Each (g, m, d) row probably carries non-trivial component-specific signal — e.g. an m6A peak in the 3'UTR vs CDS of the same gene; pA usage at site A vs site B. Aggregating them into one `(g, m)` embedding+bias throws this away.

## Constraints

- Cross-gene parameter sharing keyed by `d` (or `(m, d)`) doesn't work — see above.
- `chr:pos`-style site rows are unique per gene by construction and stay excluded from training (`RowStratum::Site` in `feature_table.rs`).
- Mixture-component rows (`m6A/comp_0..C`, `A2I/comp_0..C`, `pA/comp_0..C`) are the candidates for this work.

## Possible approaches

1. **Per-(g, m, d) bias only.** Expand `b_comp` from `(G, M)` to `(G, Σ_m C_m)`. Gives each component a per-gene-local scalar baseline magnitude. Cheap (~18k extra scalars for typical dims). Doesn't change embedding *direction*; just per-row offset. Conservative starting point.

2. **Per-(g, m, d) free offset.** Each row gets a small free vector `δ_{g, m, d} ∈ ℝ^H` added to the embedding. Lets each component shift the gene's embedding direction. Costs G·Σ_m C_m · H — substantial. Memorization-style; no cross-gene sharing.

3. **Anonymous-component reordering.** Treat each gene's components as anonymous slots, learn a soft matching between the gene's local slots and a shared modality basis of `C_m` "modes". Honors the non-identifiability constraint while still recovering cross-gene structure where it exists. Research project, not a quick refactor (optimal-transport-style soft assignment).

4. **Component-as-covariate.** Step outside the embedding-only framing: treat observed component counts as features that predict gene activity in a GLM, with per-(g, m, d) regression coefficients. Loses the "embed everything" property but is the most direct way to ask which components activate vs inhibit gene activity.

## Recommendation

Start with **(1) per-(g, m, d) bias** — cheapest, smallest implementation, validates whether component-level magnitude variation carries signal at all before investing in (2) or (3).

## Related

- Discussed in the session leading to 8a177bc.
- Aliases: `rmodem`, `embed`.
- Sim has component-level ground truth via `data-beans-sim faba`'s `alpha_{m}.parquet` files (`α_{g, m, d}` mixture weights) for direct comparison.

**Comment by YPARK** [comment]
not the right way to think about embedding...
